lib/improver/utilities/cube_units.py

```diff
@@ -32,14 +32,125 @@
 
 import numpy as np
 
+import iris
 from iris.exceptions import CoordinateNotFoundError
 
 from improver.units import DEFAULT_UNITS
 
 
+def enforce_units_and_dtypes(cubes, coords=None, enforce=True):
+    """
+    Function to check the units and datatypes of cube diagnostics and
+    coordinates against the manifest in improver.units, with option to
+    enforce or fail for non-conforming data.
+
+    Args:
+        cubes (iris.cube.Cube or iris.cube.CubeList):
+            Cube or list of cubes to be checked
+
+    Kwargs:
+        coords (list or None):
```

Comment:
I added it because it's being used in the `improver.nowcasting.accumulation` plugin where the two other functions are called.  Thinking about this I'm actually keen to replace those two calls with the new function, and perhaps make the original functions private to the `cube_units` module.  What do you think?

I'll add some unit tests for the `coords` argument.

Reply:
The idea to make the `enforce_coordinate_units_and_dtypes` and `enforce_diagnostic_units_and_dtypes` private sounds reasonable. That would help to make it clear that the `enforce_units_and_dtypes` function provides the interface that we're expecting users to use.